x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/BlockHash.java

```diff
@@ -59,42 +62,47 @@ public abstract sealed class BlockHash implements Releasable, SeenGroupIds //
     public abstract BitArray seenGroupIds(BigArrays bigArrays);
 
     /**
-     * Creates a specialized hash table that maps one or more {@link Block}s to ids.
-     * @param emitBatchSize maximum batch size to be emitted when handling combinatorial
-     *                      explosion of groups caused by multivalued fields
+     * Builds {@link BlockHash}es.
      */
-    public static BlockHash build(List<HashAggregationOperator.GroupSpec> groups, BigArrays bigArrays, int emitBatchSize) {
-        if (groups.size() == 1) {
-            return newForElementType(groups.get(0).channel(), groups.get(0).elementType(), bigArrays);
-        }
-        if (groups.size() == 2) {
-            var g1 = groups.get(0);
-            var g2 = groups.get(1);
-            if (g1.elementType() == ElementType.LONG && g2.elementType() == ElementType.LONG) {
-                return new LongLongBlockHash(bigArrays, g1.channel(), g2.channel(), emitBatchSize);
-            }
-            if (g1.elementType() == ElementType.BYTES_REF && g2.elementType() == ElementType.LONG) {
-                return new BytesRefLongBlockHash(bigArrays, g1.channel(), g2.channel(), false, emitBatchSize);
+    public record Factory(BigArrays bigArrays, PageCacheRecycler recycler, Supplier<CircuitBreaker> breaker) {
+        /**
+         * Creates a specialized hash table that maps one or more {@link Block}s to ids.
+         * @param emitBatchSize maximum batch size to be emitted when handling combinatorial
+         *                      explosion of groups caused by multivalued fields
+         */
+        public BlockHash build(List<HashAggregationOperator.GroupSpec> groups, int emitBatchSize) {
+            if (groups.size() == 1) {
```

Comment:
Here's a place where we can plug in detection of the vector API not being available. If we don't have the vector API we *could* always use `PackadVauesBlockHash` which we can make sure doesn't use the new classes.

Or, potentially, we could turn off all of ESQL if you don't have the vector API.